File: 1940-maximum-xor-for-each-query/1940-maximum-xor-for-each-query.py
class Solution:
    def getMaximumXor(self, nums: List[int], maximumBit: int) -> List[int]:
        # TC- O(n)
        # SC- O(1)
        xor = 0
        for num in nums:
            xor ^= num
        n = len(nums)
        # mask = 2 ** maximumBit - 1
        mask = (1 << maximumBit) - 1
        answer = []
        for i in range(n - 1, -1, -1):
            answer.append(xor ^ mask)
            xor = xor ^ nums[i]
        
        return answer
            

File: 1940-maximum-xor-for-each-query/1940-maximum-xor-for-each-query.java
class Solution {
    public int[] getMaximumXor(int[] nums, int maximumBit) {
        int xor = 0;

        for(int num: nums) {
            xor = xor ^ num;
        }

        int n = nums.length;
        int[] answer = new int[n];
        // int mask = (2 ** maximumBit) - 1;
        int mask = (1 << maximumBit) - 1;

        for(int i = n - 1; i >= 0; i--) {
            answer[n - i - 1] = xor ^ mask;
            xor = xor ^ nums[i];
        }

        return answer;
        
    }
}